File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/tv/series/model/TvSeriesScreenedTheatricallyEpisode.java
package dev.reuss.tmdb.domain.tv.series.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;

/**
 * TV episode that has screened theatrically.
 *
 * @param id            TMDB episode id
 * @param episodeNumber episode number
 * @param seasonNumber  season number
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record TvSeriesScreenedTheatricallyEpisode(
        int id,

        @JsonProperty("episode_number")
        int episodeNumber,

        @JsonProperty("season_number")
        int seasonNumber
) implements TmdbModel {
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/tv/series/model/TvSeriesScreenedTheatricallyEpisode.kt
package dev.reuss.tmdb.domain.tv.series.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * TV episode that has screened theatrically.
 *
 * @property id TMDB episode id
 * @property episodeNumber episode number
 * @property seasonNumber season number
 */
@JvmRecord
data class TvSeriesScreenedTheatricallyEpisode(
    @all:JsonProperty("id")
    val id: Int,

    @all:JsonProperty("episode_number")
    val episodeNumber: Int,

    @all:JsonProperty("season_number")
    val seasonNumber: Int
) : TmdbModel